pkgs/development/compilers/julia/build-julia-package.nix

```
@@ -0,0 +1,163 @@
+{ lib
+, stdenv
+, config
+, patchelf
+, julia
+, computeRequiredJuliaPackages
+, computeJuliaDepotPath
+, computeJuliaLoadPath
+, computeJuliaArtifacts
+, xvfb-run
+}:
+
+# Build an individual package
+{ fullPkgName ? "${attrs.pname}-${attrs.version}"
+
+, src
+
+, dontPatch ? false
+, patches ? []
+, patchPhase ? ""
+
+, enableParallelBuilding ? true
+# Build-time dependencies for the package, which were compiled for the system compiling this.
+, nativeBuildInputs ? []
+
+# Build-time dependencies for the package, which may not have been compiled for the system compiling this.
+, buildInputs ? []
+
+# Propagate build dependencies so in case we have A -> B -> C,
+# C can import package A propagated by B
+# Run-time dependencies for the package.
+, propagatedBuildInputs ? []
+
+# Julia packages that are required at runtime for this one.
+# These behave similarly to propagatedBuildInputs, where if
+# package A is needed by B, and C needs B, then C also requires A.
+# The main difference between these and propagatedBuildInputs is
+# during the package's installation into Julia, where all
+# requiredJuliaPackages are ALSO installed into Julia.
+, requiredJuliaPackages ? []
+
+, preBuild ? ""
+, meta ? {}
+, passthru ? {}
+
+# Julia packages can distribute 'Artifacts'.  These can be some data,
+# binary, ...  We Package 'Artifacts' as independent derivations. They
+# differ from the rest of the packages in that they don't get compiled
+# and the package providing it expect to find them in a specific location.
+, isJuliaArtifact ? false
+
+, ... } @ attrs:
+
+let
+  requiredJuliaPackages' = computeRequiredJuliaPackages requiredJuliaPackages;
+  juliaDepotPath = computeJuliaDepotPath requiredJuliaPackages';
+  juliaLoadPath = computeJuliaLoadPath requiredJuliaPackages';
+  juliaArtifacts = computeJuliaArtifacts requiredJuliaPackages';
+
+  # This step is required because when
+  # a = { test = [ "a" "b" ]; }; b = { test = [ "c" "d" ]; };
+  # (a // b).test = [ "c" "d" ];
+  # This used to mean that if a package defined extra nativeBuildInputs, it
+  # would override the ones for building a Julia package (Julia itself)
+  # causing everything to fail.
+  attrs' = builtins.removeAttrs attrs [
+    "nativeBuildInputs"
+    "buildInputs"
+    "requiredJuliaPackages"
+  ];
+
+in stdenv.mkDerivation ({
+  packageName = "${fullPkgName}";
+  # The name of the package ends up being
+  # "julia-version-package-version"
+  name = "${julia.pname}-${julia.version}-${fullPkgName}";
```

```suggestion
  pname = "${julia.pname}-${julia.version}-${attrs.pname}-${attrs.version}";
  inherit (attrs) pname version;
```
Otherwise we do not have the expected pname+version every new derivation should have.

---

@fedeinthemix I think this is also necessary because in the current state 'fullPkgName' is undefined.

---

I've changed it to (similar to Python)
```julia
  name = "${julia.pname}-${julia.version}-${attrs.pname}-${attrs.version}";
  inherit (attrs) pname version;
```